test/chisel2.py

```diff
@@ -44,6 +51,19 @@
 SET_TXT = "http://localhost:8055/set-txt"
 CLEAR_TXT = "http://localhost:8055/clear-txt"
 
+def wait_for_acme_server():
+    """Wait for directory URL set in the DIRECTORY env variable to respond"""
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
```

Comment:
Can you help me understand why `ctx.verify_mode = ssl.CERT_NONE` is needed here?

Is it possible you could poll the directory using `requests` instead of `urlopen` so that the `REQUESTS_CA_BUNDLE` env var takes effect?

Reply:
Well, you are right, it is the simplest option. I did not used it in the first place, because there is no formal mechanism (like a setuptools layout) to say that `requests` module is required for chisel script.


So it happens to be here because Certbot and its dependencies require it. However, the propability that `requests` would stop to be a transitive dependency to Certbot is equivalent to Certbot been rewritten in Java, I think we can rely on it.

Reply:
Gotcha, that rationale makes sense but I think using `requests` from `chisel2.py` as an implicit dependency is fine for now.

I'd like to find time at some point in the future to clean up the duplication between Boulder's `chisel.py`,`chisel2.py` and Pebble's copy of `chisel2.py` but that's a separate concern. I think the code complexity is high enough that it deserves a `requirements.txt` file and being treated as a first class Python project on its own.

> However, the propability that `requests` would stop to be a transitive dependency to Certbot is equivalent to Certbot been rewritten in Java, I think we can rely on it.

:laughing: I for one am excited about the new `AbstractCertbotACMEProviderInterfaceFactory`